projects/libs/core-library-typescript/src/common-library/transaction-helper.ts

```diff
@@ -0,0 +1,24 @@
+import { types } from "../common-library/types";
+import { createHash } from "crypto";
+import * as stringify from "json-stable-stringify";
+
+export class TransactionHelper implements types.Transaction {
+  header: types.TransactionHeader;
+  payload: string;
+
+  constructor(transaction: types.Transaction) {
+    this.header = transaction.header;
+    this.payload = transaction.payload;
+  }
+
+  public calculateHash(): Buffer {
+    const hash = createHash("sha256");
+    hash.update(stringify(this.header));
+    hash.update(stringify(this.payload));
+    return hash.digest();
+  }
+
+  public calculateTransactionId(): string {
```

Comment:
Raising this again: let's add a semantic type for important types (e.g., `TxID` instead of `string`).

Reply:
The problem is that as long as we use protobuf, the only way to achieve that, is by declare this type as a message wrapper for string. It creates too much overhead as well as makes any code handling with it really ugly